src/lib/moment/start-end-of.js

```diff
@@ -2,46 +2,48 @@ import { normalizeUnits } from '../units/aliases';
 
 export function startOf (units) {
     units = normalizeUnits(units);
+
+    var m = this;
     // the following switch intentionally omits break keywords
     // to utilize falling through the cases.
     switch (units) {
         case 'year':
-            this.month(0);
+            m = m.month(0);
```

Comment:
I'd say this particular function could use a lower level API to avoid unnecessary cloning. Because resetting all units lower than a given unit is the most expected operation dealing with multiple units and it would be nice if it was fast. Of course we can ship this way, and fix (improve) later.